src/lib/components/DayOpenWaterV2.svelte

```diff
@@ -0,0 +1,288 @@
+				}
+			});
+		}
+		return sorted;
+	};
+
+	$: date = dtToLDS($viewedDate);
+	$: boats = Settings.getBoats(date);
+	$: displayBuoys = sortByBoat($buoys, $boatAssignments[date]);
+
+	const loadAdminComments = async () => {
+		if (date) {
+			$adminComments[date] = await getOWAdminComments(date);
+			$adminComments = { ...$adminComments };
+		} else {
+			$adminComments[date] = [];
+		}
+	};
+	$: {
+		$viewedDate && loadAdminComments();
+	}
+	const displayValue = (buoy) => {
+		if ($boatAssignments[date] === undefined) {
+			$boatAssignments[date] = {};
+		}
+		if ($boatAssignments[date][buoy] === undefined) {
+			$boatAssignments[date][buoy] = null;
+		}
+		return $boatAssignments[date][buoy];
+	};
+
+	const getBoatCount = (schedule, assignments, boat) => {
+		let count = 0;
+		if (assignments != null) {
+			for (let buoy in assignments) {
+				if (schedule.AM[buoy] && assignments[buoy] === boat) {
+					for (let rsv of schedule.AM[buoy]) {
+						count++;
+						if (rsv.resType === 'course') {
+							count += rsv.numStudents;
+						}
+					}
+				}
+			}
+		}
+		return count;
+	};
+
+	const saveAssignments = async (e) => {
+		e.target.blur();
+		let buoy = e.target.id.split('_')[0];
+		let boat = e.target.value;
+		if (boat === 'null') {
+			boat = null;
+		}
+		if ($boatAssignments[date] == null) {
+			$boatAssignments[date] = {};
+		}
+		$boatAssignments[date][buoy] = boat;
+
+		let response = await fetch('/api/assignBuoysToBoats', {
+			method: 'POST',
+			headers: { 'Content-type': 'application/json' },
+			body: JSON.stringify({ date, assignments: $boatAssignments[date] })
+		});
+		let data = await response.json();
+		if (data.status === 'success') {
+			$boatAssignments[date] = JSON.parse(data.record.assignments);
+			$boatAssignments = { ...$boatAssignments };
+		}
+	};
+
+	const boatCountPos = (profileSrc) => {
+		if (profileSrc != null) {
+			return 'top-[50px] sm:top-[120px] md:top-[110px]';
+		} else {
+			return 'top-[100px] xs:top-[110px] lg:top-[100px]';
+		}
+	};
+
+	onMount(() => {
+		loadAdminComments();
+	});
+
+	type BuoyGrouping = {
+		buoy: Buoy;
+		boat?: string | null;
+		amReservations?: Submission[];
+		pmReservations?: Submission[];
+		amAdminComment?: string;
+		pmAdminComment?: string;
+	};
+	let buoyGroupings: BuoyGrouping[] = [];
+
+	$: {
+		const today = dtToLDS($viewedDate);
+		const todayFilter = (r: Submission) =>
+			r.date === today && r.category === 'openwater' && ['pending', 'confirmed'].includes(r.status);
+		const todaysReservations = $reservations.filter(todayFilter);
+		const comments = $adminComments[today] || [];
+		buoyGroupings = $buoys
+			.map((v) => {
+				const amComment = comments.find((c) => c.buoy === v.name && c.am_pm === 'AM');
+				const pmComment = comments.find((c) => c.buoy === v.name && c.am_pm === 'PM');
+				return {
+					buoy: v,
+					boat: $boatAssignments[today]?.[v.name!],
+					amReservations: todaysReservations.filter((r) => r.owTime === 'AM' && r.buoy === v.name),
```

Comment:
I think you want to use the result of the `getOpenwaterSchedule` function here.  Might want to combine that function with this one to reduce duplication.

Reply:
i deleted that function.